src/timer.rs

```diff
@@ -283,6 +292,18 @@ macro_rules! linked_timers {
 
                     Self { tim_primary, tim_secondary }
                 }
+
+                /// Pause counting
+                fn pause(&mut self) {
+                    self.tim_primary.cr1.modify(|_, w| w.cen().clear_bit());
+                    self.tim_secondary.cr1.modify(|_, w| w.cen().clear_bit());
+                }
+
+                /// Resume counting
+                fn resume(&mut self) {
+                    self.tim_primary.cr1.modify(|_, w| w.cen().set_bit());
+                    self.tim_secondary.cr1.modify(|_, w| w.cen().set_bit());
```

Comment:
The original code resumes the secondary timer first to prevent race conditions. Looks like something got switched in the refactor.

```suggestion
                    self.tim_secondary.cr1.modify(|_, w| w.cen().set_bit());
                    self.tim_primary.cr1.modify(|_, w| w.cen().set_bit());
```